### DISCUSSION TITLE
Events State node: allow template for entity input

### DISCUSSION THREAD

**Discussion by jj-uk** [zachowj/node-red-contrib-home-assistant-websocket#1693] [Feature Request]
The Entity input for the state note cannot be a template.  I'd like to ask if this can be added at some point.
Others nodes allow template as entity inputs, e.g. current state node.

Edit: Humm, as these's no input to this node, it's probably not possible to pass in a template. Maybe accept a flow var containing a list of entities.

**Comment by zachowj** [comment]
The `events: state` node requires static entity IDs because it registers event listeners at deployment. To achieve dynamic filtering, use an `events: all` node set to listen for the `state_changed` event, followed by a `switch` node to filter specific entity IDs based on a flow context variable. This approach will allow for flexible and dynamic entity filtering.
